matrix_client/api.py

```diff
@@ -434,13 +434,20 @@ def _send(self, method, path, content=None, query_params={}, headers={},
         if headers["Content-Type"] == "application/json":
             content = json.dumps(content)
 
-        response = requests.request(
-            method, endpoint,
-            params=query_params,
-            data=content,
-            headers=headers,
-            verify=self.validate_cert
-        )
+        response = None
+        while True:
```

Comment:
Adding blocking code the default `_send` method in the SDK is really not a nice idea. Clients using the SDK can have different concurrency models and dealing with re-tries is much better left to them.